src/neuroconv/tools/signal_processing.py

```diff
@@ -40,3 +42,31 @@ def parse_rising_and_falling_frames_from_ttl(trace: np.ndarray, num_bins: int =
     falling_frames = np.where(diff_binned_states < 0)[0]
 
     return rising_frames, falling_frames
+
+
+def synchronize_timestamps_between_systems(
+    unsynchronized_timestamps: ArrayType,
+    primary_reference_timestamps: ArrayType,
+    secondary_reference_timestamps: ArrayType,
+) -> np.ndarray:
+    """
+    One method of synchronization used to align timing information across sytems.
+
+    Involves taking timestamps recorded in a secondary system and inferring their corresponding times in a primary
+    system given a pair of reference timestamps between the two systems.
+
+    Parameters
+    ----------
+    unsynchronized_timestamps : ArrayType
+        The timestamps that need to be synchronized into the primary time basis.
+    primary_reference_timestamps : ArrayType
+        A known set of timestamps in the primary system to be used as reference.
+    secondary_reference_timestamps : ArrayType
+        A known set of timestamps in the secondary system to be used as reference.
+
+    Returns
+    -------
+    synchronized_timestamps: numpy.ndarray
+        The interpolation of the unsynchronized_timestamps into the primary reference times.
+    """
+    return np.interp(x=unsynchronized_timestamps, xp=secondary_reference_timestamps, fp=primary_reference_timestamps)
```

Comment:
Since this is a one-liner, we could just get rid of this function and move `np.interp` to within the methods

Reply:
Agreed

Reply:
(and done)